### PR TITLE
[Android] Fix Button Ripple Effect not working

### LINKED ISSUE #17642: [regression/8.0.0-preview.6.8686] Button Click Ripple Effect Not Working In Android
### Description

The button click animation does not work in my Android app .Net8 for Maui, but does work in .Net7

Using the default project for both, only in .Net7 can I see the little fade animation when I press the button.

.net8
270119662-96f25308-a7fd-4e68-84d1-9e2d4c16960b

.net7
270119657-02bf0055-bd42-4d04-9cba-945a393659e5

From what I have gathered the .net 7, the button's android platform view's background value is an instance of the RippleDrawable Class. And In the .net 8, it's an instance of the BorderDrawable Class. 

### Steps to Reproduce

Create Default Maui application (Using .Net 8)
Run default application on android.
Click counter button.

### Link to public reproduction project repository

_No response_

### Version with bug

[8.0.0-preview.6.8686](https://github.com/dotnet/maui/releases/tag/8.0.0-preview.6.8686)

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

[8.0.0-preview.5.8529](https://github.com/dotnet/maui/releases/tag/8.0.0-preview.5.8529)

### Affected platforms

Android

### Affected platform versions

_No response_

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### LINKED ISSUE #18815: [regression/8.0.3] [Android] Button ripple effect not working
### Description

In previous .NET when you click on button it by default has this ripple effect. In .NET 8 even in empty template project it doesn't seem to work.

https://github.com/dotnet/maui/assets/76904408/8cbf9adc-e612-498d-b96d-51a335d14cca

### Steps to Reproduce

1. Create new MAUI project with framework target .NET 8
2. Click the button

Expected outcome: ripple effect
 Actual outcome: nothing is happening

### Link to public reproduction project repository

_No response_

### Version with bug

8.0.3

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI, Yes, this used to work in Xamarin.Forms

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 11.0 (API 30)

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Fix Button Ripple Effect not working using a custom background. 

We use the MaterialButton instead a Button on Android, and this type use `MaterialShapeDrawable`.  MaterialShapeDrawable does not support gradients. Settings a gradient or creating a composition drawable with two layers: gradient drawable and ripple drawable create a black background. [[MaterialShapeDrawable] Unable to add gradient to MaterialShapeDrawable · Issue #1167 · material-components/material-components-android (github.com)](https://github.com/material-components/material-components-android/issues/1167)
To set the gradient we set to null the backgroundTintList property and set a custom drawable in the background. The PR apply changes in the custom drawable adding the ripple effect animation to maintain the same drawable and the MaterialButton.
 surprised 1

fix-17642

### Issues Fixed

Fixes #17642 
Fixes #18815

### CODE REVIEW

File: src/Core/src/Platform/Android/BorderDrawable.cs [hunk 1]
```diff
@@ -356,13 +479,32 @@ protected virtual void DisposeBorder(bool disposing)
...
+		}
+
 		bool HasBorder()
 		{
 			InitializeBorderIfNeeded();
 
 			return _stroke != null || _borderColor != null;
 		}
 
+		bool HasRipple()
```
Comment (jsuarezruiz) [inline src/Core/src/Platform/Android/BorderDrawable.cs:501]: Only render and run the ripple animation if set the RippleColor.

File: src/Core/src/Platform/Android/BorderDrawable.cs [hunk 2]
```diff
@@ -14,8 +18,10 @@
 
 namespace Microsoft.Maui.Platform
 {
-	public class BorderDrawable : PaintDrawable
+	public class BorderDrawable : PaintDrawable, IAnimatorUpdateListener
```
Comment (PureWeen) [inline src/Core/src/Platform/Android/BorderDrawable.cs:21]: We can't add a new interface

**Review by PureWeen — CHANGES_REQUESTED** [review]

**Comment by zippo227** [comment]
@jsuarezruiz  Is this the planned fix for https://github.com/dotnet/maui/issues/18815 as well?

**Comment by cytoph** [comment]
So this will just be closed? How can it be, that throughout a whole major version of .NET, button animations (so almost everything that's being tapped) just don't work for one of the biggest platforms of .NET MAUI (Android), and Microsoft doesn't seem to care about it. I was really looking forward to migrate from Xamarin to .NET MAUI, but with this issue the app just looks plain unprofessional and cannot be delivered to any customers as it is.

**Comment by jsuarezruiz** [comment]
> So this will just be closed? How can it be, that throughout a whole major version of .NET, button animations (so almost everything that's being tapped) just don't work for one of the biggest platforms of .NET MAUI (Android), and Microsoft doesn't seem to care about it. I was really looking forward to migrate from Xamarin to .NET MAUI, but with this issue the app just looks plain unprofessional and cannot be delivered to any customers as it is.

Reopened and is a priority for the next release.

**Comment by cytoph** [comment]
> Reopened and is a priority for the next release.

That is really, really good to hear. Just to clarify: "Next release" as in ".NET 9" or as in ".NET SR1/2"? 🤔

File: src/Core/src/Platform/Android/BorderDrawable.cs (hunk 2 above)
Comment (jsuarezruiz) [reply src/Core/src/Platform/Android/BorderDrawable.cs:21]: With `IAnimatorUpdateListener`, should go to .NET 9?. In this case, we need an alternative for .NET 8.

**Pull request closed without merging**